File: bin/seed.js
import fs from 'fs'
import glob from 'glob'
import request from 'request'

import LoopNext from 'loopnext'
let syncLoop = new LoopNext().syncLoop

let url = 'http://localhost:3001/upload'

if (process.argv[2] === 'prod') {
  url = 'https://parity-server.herokuapp.com/upload'
}

glob('test/files/*.json', (err, files) => {
  if (err) console.log(err)

  let iterations = files.length

  syncLoop(iterations, (l) => {
    let idx = l.iteration()
    let file = files[idx]
    let data = JSON.parse(fs.readFileSync(file))

    console.log(`seeding with: ${file}`)

    request.post({url: url, json: true, body: data}, (error) => {
      if (error) console.log(error)
      l.next()
    })
  })
})

File: server/seed.py
#!/usr/bin/env python

import glob, os, sys
import requests

# src = "data/test/"
src = "data/ocua_17-18/"

url = 'http://localhost:5000/upload'
if len(sys.argv) > 1 and sys.argv[1] == 'prod':
    url = 'https://parity-server.herokuapp.com/upload'

os.chdir(src)

for file in glob.glob("*.json"):
    headers = {'Accept' : 'application/json', 'Content-Type' : 'application/json'}
    r = requests.post(url, data=open(file, 'rb'), headers=headers)
    print(file, r.status_code)
